src/sentry/sentry_apps/tasks/sentry_apps.py

```diff
@@ -310,10 +311,71 @@ def _process_resource_change(
                 data[name] = serialize(instance)
 
             for installation in installations:
-                # Trigger a new task for each webhook
-                send_resource_change_webhook.delay(
-                    installation_id=installation.id, event=str(event), data=data
-                )
+
+                if _is_project_allowed(installation, instance.project_id):
+                    # Trigger a new task for each webhook
+                    send_resource_change_webhook.delay(
+                        installation_id=installation.id, event=str(event), data=data
+                    )
+
+
+def _is_project_allowed(installation: RpcSentryAppInstallation, project_id: int) -> bool:
+    service_hook = _load_service_hook(installation.organization_id, installation.id)
+    if not service_hook:
+        logger.info("send_webhooks.missing_servicehook", extra={"installation_id": installation.id})
+        return False
+    # must use 2 separate helpers for cache invalidation to work correctly in [] <-> [project1, ...] scenarios
+    return not _is_project_filtering_enabled(service_hook.id) or _does_project_filter_allow_project(
+        service_hook.id, project_id
+    )
+
+
+@cache_func_for_models(
+    [
+        (
+            ServiceHook,
+            lambda service_hook: (service_hook.organization_id, service_hook.actor_id),
+        )
+    ],
+    recalculate=False,
+)
+def _load_service_hook(organization_id: int, installation_id: int) -> ServiceHook | None:
```

Comment:
We should just update the signature here to be `organization_id: int | None`

Reply:
done